File: express/public/js/sales_invoice.js
frappe.ui.form.on('Sales Invoice', {
    refresh(frm) {
        if (frm.doc.docstatus === 0) {
            frm.page.add_menu_item(__("Submit in Background"), function() {
                frappe.call('express.events.sales_invoice.submit', {doc: frm.doc.name})
                .then(r=> {
                    frappe.show_alert("Document is submitting in background");
                })
            });
        } else if (frm.doc.docstatus === 1) {
            frm.page.add_menu_item(__("Cancel in Background"), function() {
                frappe.call('express.events.sales_invoice.cancel', {doc: frm.doc.name})
                .then(r=> {
                    frappe.show_alert("Document is canceling in background");
                })
            });
        }
        
    }
})

File: express/events/sales_invoice.py
import frappe
from frappe.utils.background_jobs import enqueue

@frappe.whitelist()
def submit(doc):
    doc = frappe.get_doc('Sales Invoice', doc)
    enqueue(doc.submit, job_name= "Submitting Invoice {}".format(doc.name), queue="long")

@frappe.whitelist()
def cancel(doc):
    doc = frappe.get_doc('Sales Invoice', doc)
    enqueue(doc.cancel, job_name= "Cancelling Invoice {}".format(doc.name), queue="long")
